extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java

```diff
@@ -339,59 +345,69 @@ public void transform(TransformationContext ctx) {
-
-                String fileName = file.getKey();
-                byte[] content = file.getValue();
-                if (fileName.endsWith(FILE_TO_UPDATE)) {
-                    content = updateApiUrl(new String(content, StandardCharsets.UTF_8), healthPath)
-                            .getBytes(StandardCharsets.UTF_8);
+            if (launch.getLaunchMode().isDevOrTest()) {
+                Path tempPath = WebJarUtil.devOrTest(curateOutcomeBuildItem, launch, artifact, HEALTH_UI_WEBJAR_PREFIX);
+                updateApiUrl(tempPath.resolve(FILE_TO_UPDATE), healthPath);
+
+                smallRyeHealthBuildProducer.produce(new SmallRyeHealthBuildItem(tempPath.toAbsolutePath().toString(),
+                        httpRootPath.adjustPath(health.ui.rootPath)));
+
+                notFoundPageDisplayableEndpointProducer
+                        .produce(new NotFoundPageDisplayableEndpointBuildItem(health.ui.rootPath + "/"));
+            } else {
+                Map<String, byte[]> files = WebJarUtil.production(curateOutcomeBuildItem, artifact, HEALTH_UI_WEBJAR_PREFIX);
+
+                for (Map.Entry<String, byte[]> file : files.entrySet()) {
+
+                    String fileName = file.getKey();
+                    byte[] content = file.getValue();
+                    if (fileName.endsWith(FILE_TO_UPDATE)) {
+                        content = updateApiUrl(new String(content, StandardCharsets.UTF_8), healthPath)
+                                .getBytes(StandardCharsets.UTF_8);
+                    }
+                    fileName = HEALTH_UI_FINAL_DESTINATION + "/" + fileName;
+
+                    generatedResourceProducer.produce(new GeneratedResourceBuildItem(fileName, content));
+                    nativeImageResourceProducer.produce(new NativeImageResourceBuildItem(fileName));
                 }
-                fileName = HEALTH_UI_FINAL_DESTINATION + "/" + fileName;
 
-                generatedResourceProducer.produce(new GeneratedResourceBuildItem(fileName, content));
-                nativeImageResourceProducer.produce(new NativeImageResourceBuildItem(fileName));
+                smallRyeHealthBuildProducer.produce(new SmallRyeHealthBuildItem(HEALTH_UI_FINAL_DESTINATION,
+                        httpRootPath.adjustPath(health.ui.rootPath)));
             }
+        }
+    }
 
-            Handler<RoutingContext> handler = recorder
-                    .uiHandler(HEALTH_UI_FINAL_DESTINATION, httpRootPath.adjustPath(health.ui.rootPath));
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    void registerGraphQLUiHandler(
+            BuildProducer<RouteBuildItem> routeProducer,
+            SmallRyeHealthRecorder recorder,
+            SmallRyeHealthRuntimeConfig runtimeConfig,
+            SmallRyeHealthBuildItem smallRyeHealthBuildItem) throws Exception {
+
+        if (shouldInclude()) {
+            Handler<RoutingContext> handler = recorder.uiHandler(smallRyeHealthBuildItem.getHealthUiFinalDestination(),
```

Comment:
Another option would be to return two different handlers depending on the runtime config. One just returning 404 and the other properly handling the requests.

Reply:
I have done this for all of them